toolkit/library/rust/shared/lib.rs

```diff
@@ -128,5 +136,6 @@ pub extern "C" fn debug_log(target: *const c_char, message: *const c_char) {
 // functions here  This function is not intended to be called at all.
 #[no_mangle]
 pub unsafe extern "C" fn uniffi_extern_hack() {
-    uniffi::extern_symbol_hack();
+    let mut status = uniffi::RustCallStatus::default();
+    uniffi::uniffi_rustbuffer_alloc(0, &mut status);
```

Comment:
This code is really ugly, but maybe it's best for now.  We could add another extern hack macro to `uniffi` to export scaffolding functions like `rustbuffer_alloc` that are shared between all components.  But that feature would probably only ever be used by moz-central.